docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,692 @@
+        // If there is only one address, we don't need to do anything special
+        if (cacheItem.Addresses.Length == 1)
+        {
+            connectAddress = cacheItem.Addresses[0];
+        }
+        else
+        {
+            int index = cacheItem.Index;
+            connectAddress = cacheItem.Addresses[index];
+            cacheItem.IncreaseIndex();
+        }
+
+        var socket = new Socket(SocketType.Stream, ProtocolType.Tcp)
+        {
+            NoDelay = true
+        };
+        try
+        {
+            await socket.ConnectAsync(connectAddress, context.DnsEndPoint.Port, cancellationToken);
+            return new NetworkStream(socket, ownsSocket: true);
+        }
+        catch
+        {
+            socket.Dispose();
+            throw;
+        }
+    }
+};
+var client = new HttpClient(handler);
+var response = await client.GetAsync(uri);
+
+class DnsCache(IPAddress[] addresses, DateTime cacheExpireTime, int index)
+{
+    public IPAddress[] Addresses { get; } = addresses;
+    public DateTime CacheExpireTime { get; } = cacheExpireTime;
+    public int Index { get; private set; } = index;
+
+    public void IncreaseIndex()
+    {
+        Index = (Index + 1) % Addresses.Length;
+    }
+}
+```
+
+### Example: Setting SocketsHttpHandler Properties
+
+SocketsHttpHandler is a powerful and flexible handler in .NET that provides advanced configuration options for managing HTTP connections. By setting various properties of SocketsHttpHandler, you can fine-tune the behavior of your HTTP client to meet specific requirements, such as performance optimization, security enhancements, and custom connection handling.
+
+Here's an example of how to configure SocketsHttpHandler with various properties and use it with HttpClient:
+
+```c#
+var cookieContainer = new CookieContainer();
+cookieContainer.Add(new Cookie("cookieName", "cookieValue"));
+
+var handler = new SocketsHttpHandler
+{
+    AllowAutoRedirect = true,
+    AutomaticDecompression = DecompressionMethods.All,
+    Expect100ContinueTimeout = TimeSpan.FromSeconds(1),
+    CookieContainer = cookieContainer,
+    Credentials = new NetworkCredential("user", "pass"),
+    MaxAutomaticRedirections = 10,
+    MaxResponseHeadersLength = 1,
+    Proxy = new WebProxy("http://proxyserver:8080"), // Don't forget to set UseProxy
+    UseProxy = true,
+    SslOptions = new SslClientAuthenticationOptions
+    {
+        RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
+        {
+            // Custom validation logic
+            return sslPolicyErrors == SslPolicyErrors.None;
+        }
+    },
+};
+
+var client = new HttpClient(handler);
+var response = await client.GetAsync(uri);
+```
+
+### Example: Change ImpersonationLevel
+
+> [!WARNING]
+> This example contains the use of reflection, which may affect the performance of your application and can be broken in future versions of .NET. Use reflection with caution and only when necessary.
```

Comment:
I'd be more in favor of documenting something along the lines of
"this functionality is platform-specific and somewhat obsolete. If you _really_ need a workaround, see https://github.com/dotnet/runtime/blob/171f1a73a9f0fa77464995bcb893a59b9b98bc3d/src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs#L1678-L1684"
and deleting this example from the doc